harmonica/_forward/_tesseroid_utils.py

```diff
@@ -423,42 +423,95 @@ def _check_points_outside_tesseroids(coordinates, tesseroids):
         This array of tesseroids must have longitude continuity and valid
         boundaries.
         Run ``_check_tesseroids`` before.
+
+    Raises
+    ------
+    ValueError
+        If any computation point falls inside any tesseroid.
     """
+    longitude, latitude, radius = coordinates
+    conflicting = _check_points_outside_tesseroids(coordinates, tesseroids)
+    if conflicting:
+        err_msg = (
+            "Found computation point(s) inside tesseroid(s). "
+            "Computation points must be outside of tesseroids.\n"
+        )
+        for i, j in conflicting:
+            west, east, south, north, bottom, top = tesseroids[j, :]
+            err_msg += (
+                f" - Computation point '({longitude[i]}, {latitude[i]}, {radius[i]})' "
+                "inside tesseroid "
+                f"'({west}, {east}, {south}, {north}, {bottom}, {top})'.\n"
+            )
+        raise ValueError(err_msg)
+
+
+@jit(nopython=True)
+def _check_points_outside_tesseroids(coordinates, tesseroids):
     longitude, latitude, radius = coordinates[:]
-    west, east, south, north, bottom, top = tuple(tesseroids[:, i] for i in range(6))
+    conflicting = []
+    for i in range(longitude.size):
+        for j in range(tesseroids.shape[0]):
+            if _check_point_inside_tesseroid(
```

Comment:
@santisoler is it worth having a function here? The function itself is a single `if` and is called inside another `if`.

Reply:
Not at all. I coded this fairly quickly and I was prototyping a little bit with parallelization of the checks. I ultimately decided to keep them in serial. I've already merged the two functions into a single one.